modules/tools/rosbag/stat_acc.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+###############################################################################
+# Copyright 2017 The Apollo Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+"""
+Stat acc in time sequence.
+Usage:
+    ./stat_acc.py bag1 bag2 ...
+"""
+
+import sys
+import os
+import datetime
+
+from rosbag.bag import Bag
+
+
+kChassisTopic = '/apollo/canbus/chassis'
+kLocalizationTopic = '/apollo/localization/pose'
+kNano = 10 ** 9
+
+
+class AccCalculator(object):
+    """Calculate mileage."""
+
+    def __init__(self):
+        """Init."""
+        self.auto_mileage = 0.0
+        self.manual_mileage = 0.0
+        self.disengagements = 0
+
+    def calculate(self, bag_file, log_file):
+        """Calculate and log acc."""
+        last_t = None
+        last_max_abs_acc = 0
+        last_speed = None
+        with Bag(bag_file, 'r') as bag:
+            for topic, msg, t in bag.read_messages(topics=[kChassisTopic]):
+                if last_t is not None:
+                    acc = (msg.speed_mps - last_speed) / (
```

Comment:
Also not sure how accurate you want the acceleration to be, Lincoln can parse acceleration in chassis_details directly from chassis, you can refer to https://github.com/ApolloAuto/apollo/blob/master/modules/canbus/proto/chassis_detail.proto for more details.